controllers/alerts/secret.go

```diff
@@ -42,42 +49,41 @@ func (r *secretReconciler) EmptyObject() client.Object {
 	return &corev1.Secret{}
 }
 
+// UpdateExistingResource checks if the secret already exists and has the correct token.
+// If it does, it returns nil. If the secret exists but the token is incorrect, it deletes the old secret
+// and creates a new one with the updated token. If the secret does not exist, it creates a new one.
+// It deletes the old secret to force Prometheus to reload the configuration.
 func (r *secretReconciler) UpdateExistingResource(ctx context.Context, cl client.Client, resource client.Object, logger logr.Logger) (client.Object, bool, error) {
 	found := resource.(*corev1.Secret)
-	modified := false
 
 	token, err := authorization.CreateToken()
```

Comment:
Is there any way to avoid it?

Create token is a heavy operation, involved reading a file, and sign the JWT.

Can we somehow not doing it in each and every reconcile loop?

For example, maybe we can add a watch or poll of the service account key file? and/or move this to a new controller, to watch the secret, and only perform this logic if the file or the secret are changed?